### ISSUE TITLE
[Feature Request]: FaceSwap optimizer

### ISSUE THREAD

**Issue by kalle07** [lllyasviel/Fooocus#3483]
### Is there an existing issue for this?

- [X] I have searched the existing issues and checked the recent builds/commits

### What would your feature do?

all is working but ...
the weight of the whole image is taken too much

### Proposed workflow

...

### Additional information

so my wish is ...
for option FaceSwap in inpaint ... run first a face-detection and than only this region is taken for the face region in the generated image, so further its useful to doo a face-detection with low-level in after ~20% of steps (in the image that is generated) to overlap these two areas more exact ...

**Comment by mashb1t** [comment]
So sorry, but I don't get it. Can you maybe rephrase your request? ^^

**Comment by kalle07** [comment]
hmmm...
ok, i write a prompt "portrait of a woman wearing a blackshirt"
and i take an image of a woman and mark it for faseswap (this woman wears a white shirt) ...
so if i generate the image mostly a white shirt apear ...
so i dont know how faceswaper works but seems the whole image is taken not only the face-region.
and my idea in adition was to recognize as fast as possible the face in the image that is generated.
if i take 60steps i think at step 20 with 30% precision (eg. yolo) can recognize the location of the face, so also there can be calculated more exactly...
it is now a little easier to understand?
"at the moment only png with real deleted regions (not black or white painted, real deleted, so only the face is visible) are working with a color of the top i write in the prompt", i can send you such an image if you like

**Comment by Chi8wah** [comment]
Yes, I also found that the content of the non-head area in the reference picture would have a significant impact on the final result, resulting in it not meeting the prompt word requirements very well. When I changed the content of the non-head area to transparent, the final result would be more in line with the prompt word requirements.

**Comment by kalle07** [comment]
long time a ago ... with comfy-ui you have much more options ;)

**Comment by Chi8wah** [comment]
> long time a ago ... with comfy-ui you have much more options ;)

I agree with your point of view, but I am using both now. I think both have their advantages and disadvantages. If fooocus can continue to improve faceswap, it will be perfect. Unfortunately, it seems to be almost unmaintained now.

**Comment by mashb1t** [comment]
@Chi8wah it indeed is unmaintained, as lllyasviel hasn't taken over development after my leave.
See https://github.com/lllyasviel/Fooocus/discussions/3504

**Comment by Chi8wah** [comment]
> [@Chi8wah](https://github.com/Chi8wah) it indeed is unmaintained, as lllyasviel hasn't taken over development after my leave. See [#3504](https://github.com/lllyasviel/Fooocus/discussions/3504)

This is a great project, thanks again for your work, it's really a pity it's become unmaintained.
